### ISSUE TITLE
Support fn::transform at root level of resources

### ISSUE THREAD

**Issue by danieljamesscott** [aws-cloudformation/cfn-python-lint#1258]
*cfn-lint version: 0.26.0*

*Description of issue.*

We use `Fn::Transform` to work around the lack of support for functions in `DeletionPolicy`, `UpdateReplacePolicy` etc. This triggers an error:

```
    "MyBucket": {
      "Type": "AWS::S3::Bucket",
      "Properties": {
        "BucketName": "myBucket"
      },
      "Fn::Transform" : {
        "Name" : "AWS::Include",
        "Parameters" : {
            "Location" : {"Fn::Sub": "s3://${MyCFIncludesBucket}/deletionPolicy/${S3DeletionPolicy}.json" }
        }
      }
    },
```

I'm not sure whether this is a specification issue or a lint check issue (I can't see any mention of functions in the spec). I also encounter an issue with the IntelliJ CloudFormation plugin which highlights the above code as an error.

**Comment by kddejong** [comment]
I'll take a look.  Thanks for submitting this issue.

**Comment by danieljamesscott** [comment]
Sorry, I looked again and this was a problem with my template. It's not a bug.
